interop/shared/src/main/scala/scalaz/zio/interop/future.scala

```diff
@@ -1,43 +1,28 @@
 package scalaz.zio
 package interop
 
-import scala.concurrent.Future
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.util.{ Failure, Success }
 
 object future {
 
   implicit class IOObjOps(private val ioObj: IO.type) extends AnyVal {
-    def fromFuture[A](ftr: () => Future[A]): IO[Throwable, A] = {
-      val _ = ftr() // avoid warning
-      ???
-    }
+    def fromFuture[A](ftr: () => Future[A])(implicit ec: ExecutionContext): IO[Throwable, A] =
+      IO.async { (cb: ExitResult[Throwable, A] => Unit) =>
+        ftr().onComplete {
+          case Success(a) => cb(ExitResult.Completed(a))
+          case Failure(t) => cb(ExitResult.Failed(t))
+        }
+      }
   }
```

Comment:
Do you want to thunk the `Future` in this case? on `ioeffect` I just pushed this PR in - i like the idea of having `IO[E, Future[A]]` - then you can lift via `point` or `now`.

Reply:
I saw it, I commented precisely saying i don't like that :slightly_smiling_face: